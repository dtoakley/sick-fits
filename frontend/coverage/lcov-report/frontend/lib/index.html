<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> frontend/lib
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">76% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>19/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>8/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.22% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="calcTotalPrice.js"><a href="calcTotalPrice.js.html">calcTotalPrice.js</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="formatMoney.js"><a href="formatMoney.js.html">formatMoney.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="68.75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 68%;"></div><div class="cover-empty" style="width:32%;"></div></div></td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="11" class="abs medium">6/11</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 13 2019 17:39:53 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
