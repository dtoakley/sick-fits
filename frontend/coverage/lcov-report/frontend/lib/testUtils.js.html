<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for frontend/lib/testUtils.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">frontend/lib</a> testUtils.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">54.55% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/11</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">18x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import casual from 'casual';
&nbsp;
// seed it so we get consistent results
casual.seed(777);
&nbsp;
const testItem = () =&gt; ({
  __typename: 'Item',
  id: 'abc123',
  price: 5000,
  user: null,
  image: 'image.jpg',
  title: 'Some title',
  description: 'Some Description',
  largeImage: 'image-large.jpg',
});
&nbsp;
const testUser = () =&gt; ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  orders: [],
  cart: [],
});
&nbsp;
const testOrderItem = () =&gt; ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: `${casual.word}.jpg`,
  title: casual.words(),
  price: 4234,
  quantity: 1,
  description: casual.words(),
});
&nbsp;
const testOrder = () =&gt; ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 40000,
  items: [testOrderItem(), testOrderItem()],
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  user: testUser(),
});
&nbsp;
const testCartItem = overrides =&gt; ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  item: testItem(),
  user: testUser(),
  ...overrides,
});
&nbsp;
class LocalStorageMock {
<span class="fstat-no" title="function not covered" >  co</span>nstructor() {
<span class="cstat-no" title="statement not covered" >    this.store = {};</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  cl</span>ear() {
<span class="cstat-no" title="statement not covered" >    this.store = {};</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ge</span>tItem(key) {
<span class="cstat-no" title="statement not covered" >    return this.store[key] || null;</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  se</span>tItem(key, value) {
<span class="cstat-no" title="statement not covered" >    this.store[key] = value.toString();</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>moveItem(key) {
<span class="cstat-no" title="statement not covered" >    delete this.store[key];</span>
  }
}
&nbsp;
export {
  LocalStorageMock,
  testItem,
  testUser,
  testCartItem,
  testOrder,
  testOrderItem,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 13 2019 17:39:53 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
