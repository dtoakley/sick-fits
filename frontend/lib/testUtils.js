import casual from 'casual';

// seed it so we get consistent results
casual.seed(777);

const testItem = () => ({
  __typename: 'Item',
  id: 'abc123',
  price: 5000,
  user: null,
  image: 'image.jpg',
  title: 'Some title',
  description: 'Some Description',
  largeImage: 'image-large.jpg',
});

const testUser = () => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  orders: [],
  cart: [],
});

const testOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: `${casual.word}.jpg`,
  title: casual.words(),
  price: 4234,
  quantity: 1,
  description: casual.words(),
});

const testOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 40000,
  items: [testOrderItem(), testOrderItem()],
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  user: testUser(),
});

const testCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  item: testItem(),
  user: testUser(),
  ...overrides,
});

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export {
  LocalStorageMock,
  testItem,
  testUser,
  testCartItem,
  testOrder,
  testOrderItem,
};
