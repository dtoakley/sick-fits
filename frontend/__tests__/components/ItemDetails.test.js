import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'

import ItemDetails, { ITEM_DETAILS_QUERY } from '../../components/ItemDetails'
import { MockedProvider } from 'react-apollo/test-utils'
import { testItem } from '../../lib/testUtils'


describe('<ItemDetails />' , () => {
    it('renders with proper data', async () => {
        const mocks = [
            {
                // when someone makes a request with this query and variable combo
                request: { query: ITEM_DETAILS_QUERY, variables: { id: '123' } },
                // return this fake data (mocked data)
                result: {
                    data: {
                        item: testItem(),
                    },
                },
            },
        ]
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ItemDetails id="123" />
            </MockedProvider>
        )
        expect(wrapper.text()).toContain('Loading...')
        await wait()
        wrapper.update()
        expect(toJSON(wrapper.find('h2'))).toMatchSnapshot()
        expect(toJSON(wrapper.find('img'))).toMatchSnapshot()
        expect(toJSON(wrapper.find('p'))).toMatchSnapshot()
    })

    it('Errors with a not found item', async () => {
        const mocks = [
            {
                request: { query: ITEM_DETAILS_QUERY, variables: { id: '123' } },
                result: {
                    errors: [{ message: 'Item not found'}]
                },
            }
        ]
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <ItemDetails id="123" />
            </MockedProvider>
        )
        await wait()
        wrapper.update()
        const item = wrapper.find('[data-test="graphql-error"]')
        expect(item.text()).toContain('Item not found')
        expect(toJSON(item)).toMatchSnapshot()
    })
})


