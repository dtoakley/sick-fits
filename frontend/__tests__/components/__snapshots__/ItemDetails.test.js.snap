// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetails /> Errors with a not found item 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Woops!
  </strong>
  Item not found
</p>
`;

exports[`<ItemDetails /> renders with proper data 1`] = `
<h2>
  Viewing 
  Some title
</h2>
`;

exports[`<ItemDetails /> renders with proper data 2`] = `
<img
  alt="Some title"
  src="image-large.jpg"
/>
`;

exports[`<ItemDetails /> renders with proper data 3`] = `
<p>
  Some Description
</p>
`;
